tests/test_interventions.py

```diff
@@ -2084,6 +2303,186 @@ def test_test_sensitivity(self, test_params ):
+        p_val     = binom.cdf( true_neg, ( true_neg + false_pos ), test_params[ "lateral_flow_test_specificity"] )
+        np.testing.assert_equal( true_neg > 100, True, "In-sufficient true negatives cases to test" )
+        np.testing.assert_equal( false_pos > 50, True, "In-sufficient false positives cases to test" )
+        np.testing.assert_equal( p_val > lower_CI, True, f"Too few false positives given the test specificity tn={true_neg}, fp={false_pos}, p={p_val}" )
+        np.testing.assert_equal( p_val < upper_CI, True, f"Too many false positives given the test specificity tn={true_neg}, fp={false_pos}, p={p_val}" )
+
+        # check the sensitivity is monotonic on each side of the peak.
+        df_test[ "test_sensitive_inf" ]  = ( ( df_test[ "time_infected" ] != - 1 ) & ( df_test[ "time_infected" ] <= df_test[ "time" ] ) )
+        df_test[ "time_since_inf" ]  = ( df_test[ "time" ] - df_test[ "time_infected" ] )
+
+        true_pos   = df_test[ ( df_test[ "infected" ] == True ) & ( df_test[ "lateral_flow_status" ] == 1 ) & ( df_test[ "test_sensitive_inf" ] == True ) ].groupby( [ "time_since_inf" ] ).size()
+        false_neg  = df_test[ ( df_test[ "infected" ] == True ) & ( df_test[ "lateral_flow_status" ] == 0 ) & ( df_test[ "test_sensitive_inf" ] == True ) ].groupby( [ "time_since_inf" ] ).size()
+        sens_ratio = true_pos.divide( false_neg.add( true_pos, fill_value=0 ), fill_value=0 )
+        sens_peak  = sens_ratio.argmax()
+
+        sens_diff = sens_ratio.diff()
+        is_sens_single_peak = np.all( sens_diff.iloc[ 2 : sens_peak + 1 ] >= 0 ) & np.all( sens_diff[ sens_peak + 1 : -3 ] <= 0 )
+        np.testing.assert_equal( is_sens_single_peak, True, f"Sensitivity does not have a single peak: {sens_diff}" )
+
+        # check the sensitivity at the peak.
+        df_test[ "symptomatic" ] = ( df_test["time_symptomatic"] > 0 )
+        true_pos   = df_test[ ( df_test[ "time_since_inf" ] == 3 ) & ( df_test[ "symptomatic" ] == True ) & ( df_test[ "lateral_flow_status" ] == 1 ) & ( df_test[ "test_sensitive_inf" ] == True ) ].groupby( [ "time_since_inf" ] ).size()
+        false_neg  = df_test[ ( df_test[ "time_since_inf" ] == 3 ) & ( df_test[ "symptomatic" ] == True ) & ( df_test[ "lateral_flow_status" ] == 0 ) & ( df_test[ "test_sensitive_inf" ] == True ) ].groupby( [ "time_since_inf" ] ).size()
+        sens_ratio = true_pos.divide( false_neg.add( true_pos, fill_value=0 ), fill_value=0 )
+        sens_peak_idx  = sens_ratio.idxmax()
+
+        np.testing.assert_equal( false_neg[ sens_peak_idx ] > 100, True, "In-sufficient false negatives in sensitive period to test" )
+        np.testing.assert_equal( true_pos[ sens_peak_idx ] > 100, True, "In-sufficient true positives in sensitive period to test" )
+
+        sd_mult = test_params[ "sd_infectiousness_multiplier" ]
+        sens_exp = test_params[ "lateral_flow_test_sensitivity"]
+        if sd_mult:
+          sens_low = sens_exp - 0.675 / 2.0 * sd_mult
```

Comment:
Where do these numbers come from? Hard to see from the code in `interventions.c`.